File: 3024. Type of Triangle/soln.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return String
     */
    function triangleType($nums) {
        sort($nums);
        if(!$this->isTriangle($nums)) return "none";
        if($this->isEquilateral($nums)) return "equilateral";
        if($this->isIsosceles($nums)) return "isosceles";
        return "scalene";
    }
    private function isTriangle($nums){
        return $nums[0] + $nums[1] > $nums[2] &&
        $nums[0] + $nums[2] > $nums[1] &&
        $nums[1] + $nums[2] > $nums[0];
    }
    private function isEquilateral($nums){
        return $nums[0] == $nums[1] && $nums[1] == $nums[2];
    }

    private function isIsosceles($nums){
        return $nums[0] == $nums[1] || $nums[1] == $nums[2] || $nums[0] == $nums[2];
    }
}

File: Math/3024. Type of Triangle/soln.java
/*
class Solution {
    public String triangleType(int[] nums) {
        Arrays.sort(nums);
        if (nums[0] + nums[1] <= nums[2])
            return "none";
        if (nums[0] == nums[1] && nums[1] == nums[2]) {
            return "equilateral";
        }
        if (nums[0] == nums[1] || nums[1] == nums[2]) {
            return "isosceles";
        }
        return "scalene";
    }
}
*/

import java.util.*;

//faster
class Solution {
    public String triangleType(int[] nums) {
        if (!isTriangle(nums)) return "none";
        if (isEquilateral(nums)) return "equilateral";
        if (isIsosceles(nums)) return "isosceles";
        return "scalene";
    }

    private boolean isTriangle(int[] nums) {
        return nums[0] + nums[1] > nums[2] &&
               nums[0] + nums[2] > nums[1] &&
               nums[1] + nums[2] > nums[0];
    }

    private boolean isEquilateral(int[] nums) {
        return nums[0] == nums[1] && nums[1] == nums[2];
    }

    private boolean isIsosceles(int[] nums) {
        return nums[0] == nums[1] || nums[1] == nums[2] || nums[0] == nums[2];
    }
}

public class soln{
    public static void main(String[] args){
        Solution sol = new Solution();
        Scanner sc = new Scanner(System.in);
        int[] nums = new int[3];
        for (int i = 0; i < 3; i++) {
            nums[i] = sc.nextInt();
        }
        String ans = sol.triangleType(nums);
        System.out.println(ans);
    }
}

